code/obj/item/clothing/chameleon_js.dm

```diff
@@ -269,3 +272,1557 @@
+
+	engineer
+		name = "Engineer"
+		jumpsuit_type = new/datum/chameleon_jumpsuit_pattern/rank/engineer
+		hat_type = new/datum/chameleon_hat_pattern/hardhat
+		suit_type = new/datum/chameleon_suit_pattern/winter_coat_engineering
+		glasses_type = new/datum/chameleon_glasses_pattern/meson
+		shoes_type = new/datum/chameleon_shoes_pattern/orange
+		gloves_type = new/datum/chameleon_gloves_pattern/insulated
+		belt_type = new/datum/chameleon_belt_pattern
+		backpack_type = new/datum/chameleon_backpack_pattern/engineer
+
+	mechanic
+		name = "Mechanic"
+		jumpsuit_type = new/datum/chameleon_jumpsuit_pattern/rank/mechanic
+		hat_type = new/datum/chameleon_hat_pattern/hardhat
+		suit_type = new/datum/chameleon_suit_pattern/winter_coat_engineering
+		glasses_type = new/datum/chameleon_glasses_pattern/meson
+		shoes_type = new/datum/chameleon_shoes_pattern/brown
+		gloves_type = new/datum/chameleon_gloves_pattern/insulated
+		belt_type = new/datum/chameleon_belt_pattern
+		backpack_type = new/datum/chameleon_backpack_pattern/engineer
+
+	miner
+		name = "Miner"
+		jumpsuit_type = new/datum/chameleon_jumpsuit_pattern/rank/overalls
+		hat_type = new/datum/chameleon_hat_pattern/space_helmet_engineer
+		suit_type = new/datum/chameleon_suit_pattern/space_suit_engineering
+		glasses_type = new/datum/chameleon_glasses_pattern/meson
+		shoes_type = new/datum/chameleon_shoes_pattern/orange
+		gloves_type = new/datum/chameleon_gloves_pattern
+		belt_type = new/datum/chameleon_belt_pattern/miner
+		backpack_type = new/datum/chameleon_backpack_pattern/engineer
+
+
+	rancher
+		name = "Rancher"
+		jumpsuit_type = new/datum/chameleon_jumpsuit_pattern/rank/rancher
+		hat_type = new/datum/chameleon_hat_pattern/cowboy_hat
+		suit_type = new/datum/chameleon_suit_pattern/botanist_apron
+		glasses_type = new/datum/chameleon_glasses_pattern
+		shoes_type = new/datum/chameleon_shoes_pattern/brown
+		gloves_type = new/datum/chameleon_gloves_pattern
+		belt_type = new/datum/chameleon_belt_pattern
+		backpack_type = new/datum/chameleon_backpack_pattern
+
+	botanist
+		name = "Botanist"
+		jumpsuit_type = new/datum/chameleon_jumpsuit_pattern/rank/hydroponics
+		hat_type = new/datum/chameleon_hat_pattern/cowboy_hat
+		suit_type = new/datum/chameleon_suit_pattern/botanist_apron
+		glasses_type = new/datum/chameleon_glasses_pattern/sunglasses
+		shoes_type = new/datum/chameleon_shoes_pattern/brown
+		gloves_type = new/datum/chameleon_gloves_pattern
+		belt_type = new/datum/chameleon_belt_pattern
+		backpack_type = new/datum/chameleon_backpack_pattern
+
+	janitor
+		name = "Janitor"
+		jumpsuit_type = new/datum/chameleon_jumpsuit_pattern/rank/janitor
+		hat_type = new/datum/chameleon_hat_pattern/janiberet
+		suit_type = new/datum/chameleon_suit_pattern/bio_suit
+		glasses_type = new/datum/chameleon_glasses_pattern
+		shoes_type = new/datum/chameleon_shoes_pattern/galoshes
+		gloves_type = new/datum/chameleon_gloves_pattern/latex
```

Comment:
Replace these with the cleaning gloves. I'm not sure if janitors even have latex gloves in their rooms on all maps.

Reply:
I meant to do this and forgot to, fixed, thanks!